module/VuFind/tests/unit-tests/src/VuFindTest/Db/Service/SessionServiceTest.php

```diff
@@ -0,0 +1,319 @@
+                ->willReturn($session);
+        }
+        return $serviceMock;
+    }
+
+    /**
+     * Test retriving an session object from database.
+     *
+     * @return void
+     */
+    public function testGetSessionById()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getQueryBuilder('1', [$session]);
+        $entityManager->expects($this->once())->method('createQueryBuilder')
+            ->willReturn($queryBuilder);
+        $service = $this->getService($entityManager, $pluginManager);
+        $this->assertEquals($session, $service->getBySessionId('1', false));
+    }
+
+    /**
+     * Test creating a new session if no existing session is found.
+     *
+     * @return void
+     */
+    public function testCreatingSession()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session, 1);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getQueryBuilder('1', []);
+        $entityManager->expects($this->once())->method('createQueryBuilder')
+            ->willReturn($queryBuilder);
+        $session->expects($this->once())->method('setSessionId')
+            ->with($this->equalTo('1'))
+            ->willReturn($session);
+        $session->expects($this->once())->method('setCreated')
+            ->with($this->anything())
+            ->willReturn($session);
+        $service = $this->getService($entityManager, $pluginManager, $session);
+        $this->assertEquals($session, $service->getBySessionId('1', true));
+    }
+
+    /**
+     * Test reading session data.
+     *
+     * @return void
+     */
+    public function testReadSession()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session, 1);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getQueryBuilder('1', [$session]);
+        $entityManager->expects($this->once())->method('createQueryBuilder')
+            ->willReturn($queryBuilder);
+        $session->expects($this->once())->method('getLastUsed')
+            ->willReturn(time() - 1000);
+        $session->expects($this->once())->method('setLastUsed')
+            ->with($this->anything());
+        $session->expects($this->once())->method('getData')
+            ->willReturn('foo');
+        $service = $this->getService($entityManager, $pluginManager);
+        $this->assertEquals('foo', $service->ReadSession('1', 10000000));
```

Comment:
Maybe it's worth adding another test that tries to read a session that is expired -- i.e. getLastUsed returns a very low number.